docs/learn/learn-comparison-ethereum-2.md

```diff
@@ -1,191 +1,322 @@
-with a suite of modules that can be configured, composed, and extended to develop a chain's STF.
+should be a similar set of development tools like Truffle and Ganache to develop for eWasm. See
+[eWasm Design](https://github.com/ewasm/design) for more information. See
+[Sharding FAQ](https://github.com/ethereum/wiki/wiki/Sharding-FAQ#how-would-synchronous-cross-shard-messages-work)
+for more information.
+
+Every shard in {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} has an abstract STF
+based on Wasm. Each shard can expose a custom interface as long as the logic compiles to Wasm and
+the shard provides an "execute block" function to
+{{ polkadot: Polkadot's :polkadot }}{{ kusama: Kusama's :kusama }} validators. Polkadot’s
+[Substrate](https://substrate.io/) development framework allows full spectrum composability with a
+suite of modules that can be configured, composed, and extended to develop a chain's STF.
+
+## Rollups
+
+Rollups are a layer-2 scaling solution. A layer-2 network of nodes would be responsible for "rolling
+up" transactions by batching them before publishing them to the main Ethereum chain. A rolled-up
+transaction could include thousands of transactions. Rollups increase TPS and can reduce transaction
+fees significantly. Rollups are a broad category of solutions encompassing many possible
+architectures and implementations. Rollups on Ethereum are similar to
+{{ polkadot: Polkadot's :polkadot }}{{ kusama: Kusama's :kusama }} parachains, where you prove some
+set of transactions happened on another "mini-chain".
+
+Note that the bitcoin Lightning Network at a high levels allows to achieve similar goals (i.e.
+unload unnecessary storage on chain), but it is not a rollup. The Lightning Network lets you lock
+bitcoin up on-chain, then puts it in channels, that are all shared between different nodes. Bitcoin
+can then be routed across these channels. There's no "lightning blockchain".
+
+### Optimistic Rollups
+
+Using an "innocent until proven guilty" approach, optimistic rollups will accept transaction
+commitments to the network while validating for errors and security issues. If something is invalid,
+validators can submit fraudulent commitment proof (fraud-proofs). These proofs will be used to
+reject fraudulent transactions. This way, transaction results and proofs can be stored on the main
+network, but transaction data can be stored elsewhere. Optimistic rollups can improve transaction
+scalability 10-100x and write transactions to Ethereum as
+[`calldata`](https://ethereum.stackexchange.com/questions/52989/what-is-calldata), reducing gas fees
+significantly. Examples of optimistic rollup solutions include
+[Optimisim](https://www.optimism.io/), [Arbitrum](https://bridge.arbitrum.io/) and
+[Unipig](https://unipig.exchange/welcome). See
+[ETH Docs: Optimistic Rollups](https://ethereum.org/en/developers/docs/scaling/optimistic-rollups/)
+for more information.
+
+### Zero-knowledge Rollups
```

Comment:
I would shorten this, add new info, and make it its own h2 section with what ETH 2.0 and Polkadot are doing underneath respectively.  Might be wise to include some of the parachains who are working with ZK proofs in here as well?